static/js/publisher/builds/components/repoConnect.js

```diff
@@ -0,0 +1,348 @@
+import React, { Fragment } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+
+import Button from "@canonical/react-components/dist/components/Button";
+import DatalistSelect from "./datalistSelect";
+import Select from "./select";
+
+class RepoConnect extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedRepo: "",
+      repoList: [],
+      selectedOrganization: "",
+      organizations: this.props.organizations,
+      user: this.props.user,
+      isRepoListDisabled: true,
+      message: { icon: "", description: "" },
```

Comment:
I'd feel more comfortable with this being:
```
message: <string>,
status: <type>,
```

Where we can set the "type" in some consts like:
```
const LOADING = "LOADING";
const ERROR = "ERROR";
const NAME_MISMATCH = "NAME_MISMATCH";
const SUCCESS = "SUCCESS";
```

Then in the code we can set:
- for default state
```
message: null,
status: null
```
- for any other state
```
message: "The message"|null,
status: LOADING|ERROR|NAME_MISMATCH|SUCCESS,
```